Easydict/Feature/Service/Gemini/GeminiService.swift

```diff
@@ -0,0 +1,94 @@
+//
+//  GeminiService.swift
+//  Easydict
+//
+//  Created by Jerry on 2024-01-02.
+//  Copyright © 2024 izual. All rights reserved.
+//
+
+import Foundation
+import GoogleGenerativeAI
+
+@objc(EZGeminiService)
+public final class GeminiService: QueryService {
+    override public func serviceType() -> ServiceType {
+        .gemini
+    }
+
+    override public func link() -> String? {
+        "https://bard.google.com/chat"
+    }
+
+    override public func name() -> String {
+        NSLocalizedString("gemini_translate", comment: "The name of Gemini Translate")
+    }
+
+    override public func supportLanguagesDictionary() -> MMOrderedDictionary<AnyObject, AnyObject> {
+        // TODO: Replace MMOrderedDictionary.
+        let orderedDict = MMOrderedDictionary<AnyObject, AnyObject>()
+        GeminiTranslateType.supportLanguagesDictionary.forEach { key, value in
+            orderedDict.setObject(value as NSString, forKey: key.rawValue as NSString)
+        }
+        return orderedDict
+    }
+
+    override public func ocr(_: EZQueryModel) async throws -> EZOCRResult {
+        NSLog("Gemini Translate does not support OCR")
+        throw QueryServiceError.notSupported
+    }
+
+    override public func needPrivateAPIKey() -> Bool {
+        true
+    }
+
+    override public func hasPrivateAPIKey() -> Bool {
+        if apiKey == defaultAPIKey {
+            return false
+        }
+        return true
+    }
+
+    private let defaultAPIKey = "" /* .decryptAES() */
+
+    // easydict://writeKeyValue?EZGeminiAPIKey=xxx
+    private var apiKey: String {
+        let apiKey = UserDefaults.standard.string(forKey: EZGeminiAPIKey)
+        if let apiKey, !apiKey.isEmpty {
+            return apiKey
+        } else {
+            return defaultAPIKey
+        }
+    }
+
+    override public func autoConvertTraditionalChinese() -> Bool {
+        true
+    }
+
+    override public func translate(_ text: String, from: Language, to: Language, completion: @escaping (EZQueryResult, Error?) -> Void) {
+        Task {
+            // https://github.com/google/generative-ai-swift
+            do {
+                var resultString = ""
+                let prompt = "You are a translation expert proficient in various languages that can only translate text and cannot interpret it. You are able to accurately understand the meaning of proper nouns, idioms, metaphors, allusions or other obscure words in sentences and translate them into appropriate words by combining the context and language environment. The result of the translation should be natural and fluent, you can only return the translated text, do not show additional information and notes. Translate the following \(from.rawValue) text into \(to.rawValue): \(text)"
```

Comment:
Maybe we can consider refactoring the sentence before "Translate the following \(from.rawValue) text into \(to.rawValue): \(text)" to a static string property in GeminiService.